--- redland_wrap.c.orig	2004-01-14 18:14:13.000000000 +0000
+++ redland_wrap.c	2004-01-14 18:14:13.000000000 +0000
@@ -1848,7 +1848,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value(arg1);
-    
+    if(!result)
+      result="";
     
     ZVAL_STRING(return_value,result, 1);
     
@@ -1875,6 +1876,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_as_latin1(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -1902,6 +1905,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_language(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -4485,6 +4490,7 @@
 
 PHP_MSHUTDOWN_FUNCTION(redland)
 {
+    librdf_php_world_finish();
     return SUCCESS;
 }
 PHP_MINIT_FUNCTION(redland)
