--- redland_wrap.c	2006-03-02 17:23:31.000000000 -0800
+++ redland_wrap.c.new	2006-03-02 17:22:59.000000000 -0800
@@ -7113,6 +7113,7 @@
 {
 /* shutdown section */
 
+    librdf_php_world_finish();
     return SUCCESS;
 }
 PHP_RSHUTDOWN_FUNCTION(redland)
@@ -7123,5 +7124,10 @@
 }
 PHP_MINFO_FUNCTION(redland)
 {
+   php_info_print_table_start();
+   php_info_print_table_header(2, "Redland RDF Support", "enabled");
+   php_info_print_table_row(2, "redland librdf version", librdf_version_string);
+   php_info_print_table_row(2, "redland php bindings version", REDLAND_BINDINGS_VERSION);
+   php_info_print_table_end();
 }
 /* end init section */
