--- redland_wrap.c.orig	2004-09-29 15:36:58.592057014 +0100
+++ redland_wrap.c	2004-09-29 15:36:58.646049920 +0100
@@ -1647,7 +1647,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value(arg1);
-    
+    if(!result)
+      result="";
     
     ZVAL_STRING(return_value,result, 1);
     
@@ -1674,6 +1675,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_as_latin1(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -1701,6 +1704,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_language(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -5159,6 +5164,7 @@
 
 PHP_MSHUTDOWN_FUNCTION(redland)
 {
+    librdf_php_world_finish();
     return SUCCESS;
 }
 PHP_MINIT_FUNCTION(redland)
