--- redland_wrap.c.orig	2004-11-02 19:24:48.496789437 +0000
+++ redland_wrap.c	2004-11-02 19:24:48.525785630 +0000
@@ -1727,7 +1727,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value(arg1);
-    
+    if(!result)
+      result="";
     
     ZVAL_STRING(return_value,result, 1);
     
@@ -1754,6 +1755,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_as_latin1(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -1781,6 +1784,8 @@
     }
     
     result = (char *)librdf_node_get_literal_value_language(arg1);
+    if(!result)
+      result="";
     
     
     ZVAL_STRING(return_value,result, 1);
@@ -5327,6 +5332,7 @@
 
 PHP_MSHUTDOWN_FUNCTION(redland)
 {
+    librdf_php_world_finish();
     return SUCCESS;
 }
 PHP_MINIT_FUNCTION(redland)
