--- redland_wrap.c.orig	2005-05-23 15:40:52.000000000 +0100
+++ redland_wrap.c	2005-05-23 15:40:52.000000000 +0100
@@ -680,7 +680,7 @@
   p = value->ptr;
   if (type==-1) return -1;
 
-  type_name=zend_rsrc_list_get_rsrc_type(z->value.lval);
+  type_name=zend_rsrc_list_get_rsrc_type(z->value.lval TSRMLS_CC);
 
   return SWIG_ZTS_ConvertResourceData(p,type,type_name,ptr,ty TSRMLS_CC);
 }
@@ -6754,6 +6754,7 @@
 
 PHP_MSHUTDOWN_FUNCTION(redland)
 {
+    librdf_php_world_finish();
     return SUCCESS;
 }
 PHP_MINIT_FUNCTION(redland)
@@ -6826,5 +6827,10 @@
 }
 PHP_MINFO_FUNCTION(redland)
 {
+    php_info_print_table_start();
+    php_info_print_table_header(2, "Redland RDF Support", "enabled");
+    php_info_print_table_row(2, "redland librdf version", librdf_version_string);
+    php_info_print_table_row(2, "redland php bindings version", REDLAND_BINDINGS_VERSION);
+    php_info_print_table_end();
 }
 /* end init section */
